mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,140 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
That works, and it's what I had in an earlier PR, but see discussion above (TLDR: r^2, MSE and other small stats might be useful to have especially for model save/load, marking this transient will make those stats inaccessible).